packages/jest-runtime/src/index.ts

```diff
@@ -158,6 +159,7 @@ class Runtime {
     this._sourceMapRegistry = Object.create(null);
     this._fileTransforms = new Map();
     this._virtualMocks = Object.create(null);
+    this._requireStack = [];
```

Comment:
I don't think the stack should be per runtime, but per module.

Looking at node's implementation, they construct the array at runtime when not finding a module by walking up the `module.parent` property - I think we should do the same.

https://github.com/nodejs/node/blob/ecfb7b0988da93b1b140f7fbe877824296697364/lib/internal/modules/cjs/loader.js#L1020-L1034